plugin/src/main/java/lol/pyr/znpcsplus/storage/sqlite/SQLiteStorage.java

```diff
@@ -0,0 +1,317 @@
+package lol.pyr.znpcsplus.storage.sqlite;
+
+import lol.pyr.znpcsplus.api.entity.EntityProperty;
+import lol.pyr.znpcsplus.config.ConfigManager;
+import lol.pyr.znpcsplus.database.SQLite;
+import lol.pyr.znpcsplus.entity.EntityPropertyImpl;
+import lol.pyr.znpcsplus.entity.EntityPropertyRegistryImpl;
+import lol.pyr.znpcsplus.entity.PropertySerializer;
+import lol.pyr.znpcsplus.hologram.HologramImpl;
+import lol.pyr.znpcsplus.interaction.ActionRegistry;
+import lol.pyr.znpcsplus.npc.NpcEntryImpl;
+import lol.pyr.znpcsplus.npc.NpcImpl;
+import lol.pyr.znpcsplus.npc.NpcTypeRegistryImpl;
+import lol.pyr.znpcsplus.packets.PacketFactory;
+import lol.pyr.znpcsplus.storage.NpcStorage;
+import lol.pyr.znpcsplus.util.NpcLocation;
+import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
+
+import java.io.File;
+import java.math.BigDecimal;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class SQLiteStorage implements NpcStorage {
+    private final static Logger logger = Logger.getLogger("SQLiteStorage");
+
+    private final PacketFactory packetFactory;
+    private final ConfigManager configManager;
+    private final ActionRegistry actionRegistry;
+    private final NpcTypeRegistryImpl typeRegistry;
+    private final EntityPropertyRegistryImpl propertyRegistry;
+    private final LegacyComponentSerializer textSerializer;
+    private final SQLite database;
+
+    private final String TABLE_NPCS;
+    private final String TABLE_NPCS_PROPERTIES;
+    private final String TABLE_NPCS_HOLOGRAMS;
+    private final String TABLE_NPCS_ACTIONS;
+
+    public SQLiteStorage(PacketFactory packetFactory, ConfigManager configManager, ActionRegistry actionRegistry, NpcTypeRegistryImpl typeRegistry, EntityPropertyRegistryImpl propertyRegistry, LegacyComponentSerializer textSerializer, File file) {
+        this.packetFactory = packetFactory;
+        this.configManager = configManager;
+        this.actionRegistry = actionRegistry;
+        this.typeRegistry = typeRegistry;
+        this.propertyRegistry = propertyRegistry;
+        this.textSerializer = textSerializer;
+        this.database = new SQLite(file, logger);
+        database.load();
+        if (database.getSQLConnection() == null) {
+            throw new RuntimeException("Failed to initialize SQLite Storage.");
+        }
+        TABLE_NPCS = "npcs";
+        TABLE_NPCS_PROPERTIES = "npcs_properties";
+        TABLE_NPCS_HOLOGRAMS =  "npcs_holograms";
+        TABLE_NPCS_ACTIONS = "npcs_actions";
+        validateTables();
+    }
+
+    private void validateTables() {
+        if (!database.tableExists(TABLE_NPCS)) {
+            logger.log(java.util.logging.Level.INFO, "Creating table " + TABLE_NPCS + "...");
```

Comment:
Can we use `Logger#info(String)` instead of providing the log level manually please?

Reply:
Sure

Reply:
fixed in eb414ce